WSCoachMarksView.m

```diff
@@ -198,12 +215,45 @@ - (void)goToCoachMarkIndexed:(NSUInteger)index {
     // Coach mark definition
     NSDictionary *markDef = [self.coachMarks objectAtIndex:index];
     NSString *markCaption = [markDef objectForKey:@"caption"];
-    CGRect markRect = [[markDef objectForKey:@"rect"] CGRectValue];
     
     NSString *shape = @"other";
     if([[markDef allKeys] containsObject:@"shape"])
         shape = [markDef objectForKey:@"shape"];
     
+    UIEdgeInsets markInsets = UIEdgeInsetsZero;
+    if ([[markDef allKeys] containsObject:@"insets"])
+        markInsets = [[markDef objectForKey:@"insets"] UIEdgeInsetsValue];
+    
+    // Type-check the object found for the "views" key
+    NSArray *markViewsArray = nil;
+    id viewOrArray = [markDef objectForKey:@"views"];
+    if ([viewOrArray isKindOfClass:[UIView class]])
+    {
+        markViewsArray = @[viewOrArray];
+    }
+    else if ([viewOrArray isKindOfClass:[NSArray class]])
+    {
+        markViewsArray = viewOrArray;
+    }
```

Comment:
You're going to want put the `else` here to deal with either `views` being absent, or `views` holding something that isn't an array or a dictionary.